clientimpl.go

```diff
@@ -34,216 +32,239 @@ func NewMazzarothClient(servers []string, options ...Options) (*ClientImpl, erro
+		return xdrResp.Receipt, nil
+	}
+
+	return nil, errors.New("Missing receipt")
 }
 
-// AccountLookup calls the endpoint: /v1/channels/{channel_id}/accounts/{accout_id}.
-func (c *ClientImpl) AccountLookup(channelID string, seed string) (*xdr.Response, error) {
-	seedBin, err := hex.DecodeString(seed)
+// TransactionSubmit calls the endpoint: /v1/channels/{channel_id}/transactions.
+func (c *ClientImpl) TransactionSubmit(ctx context.Context, transaction *xdr.Transaction) (*xdr.ID, *xdr.Receipt, error) {
+	b, err := json.Marshal(transaction)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to decode seed")
+		return nil, nil, errors.Wrap(err, "unable to marshal to json")
 	}
-	privateKey := ed25519.NewKeyFromSeed(seedBin)
-	address, err := xdr.IDFromPublicKey(privateKey.Public())
+
+	channelID := hex.EncodeToString(transaction.Data.ChannelID[:])
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to create private key")
+		return nil, nil, errors.Wrap(err, "unable to unmarshal the channelID")
 	}
 
-	url := fmt.Sprintf("%s/%s/channels/%s/accounts/%s", c.serverSelector.Pick(), version, channelID, hex.EncodeToString(address[:]))
+	url := fmt.Sprintf("%s/%s/channels/%s/transactions", c.address, version, channelID)
 
-	response, err := makeRequest(c.httpClient, http.MethodGet, url, nil)
+	xdrResp, err := c.do(ctx, url, http.MethodPost, b)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to make a request to channel lookup endpoint")
+		return nil, nil, errors.Wrap(err, "unable to make a request to transaction submit endpoint")
 	}
 
-	return response, nil
+	if xdrResp.TransactionID != nil {
+		return xdrResp.TransactionID, nil, nil
+	} else if xdrResp.Receipt != nil {
+		return &xdrResp.Receipt.TransactionID, xdrResp.Receipt, nil
+	}
+	return nil, nil, errors.New("unable to processes transaction")
 }
 
-func makeRequest(httpClient *http.Client, method, url string, body io.Reader) (*xdr.Response, error) {
-	fmt.Println("--------------------------------------------------------------------------------------------------------")
-	fmt.Println(url)
+// TransactionLookup calls the endpoint: /v1/channels/{channel_id}/transactions/{id}.
+func (c *ClientImpl) TransactionLookup(ctx context.Context, channelID string, transactionID string) (*xdr.Transaction, error) {
+	url := fmt.Sprintf("%s/%s/channels/%s/transactions/%s", c.address, version, channelID, transactionID)
 
-	req, err := http.NewRequestWithContext(context.Background(), method, url, body)
+	xdrResp, err := c.do(ctx, url, http.MethodGet, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to handle http response")
+	}
+
+	if xdrResp.Transaction != nil {
+		return xdrResp.Transaction, nil
+	}
+
+	return nil, errors.New("missing transaction")
+}
+
+func (c *ClientImpl) do(ctx context.Context, url string, method string, body []byte) (*xdr.Response, error) {
+	req, err := http.NewRequestWithContext(context.Background(), method, url, bytes.NewReader(body))
```

Comment:
Shouldn't this pass the function context here instead of creating a new one?

Reply:
yes thanks - copy past from the old method